gateway/src/apicast/policy/url_rewriting/url_rewriting.lua

```diff
@@ -28,6 +28,30 @@ local function substitute(func, subject, regex, replace, options)
   return new_uri, num_changes > 0
 end
 
+-- Returns true if the Method of the request is in the methods of the command meaning the rewrite rule should be applied
+-- Returns true if no Method is provided in the config for backwardscompatibility
+local function should_apply(methods)
+  local request_method = ngx.req.get_method()
+  local next = next
+
+  local should_apply_command = true
+
+  if methods ~= nil and next(methods) ~= nil  then
```

Comment:
This code can be a bit simpler, what about: 

```
local function should_apply(methods)
  local request_method = ngx.req.get_method()

  if methods == nil and next(methods) == nil  then
    return true
  end

  for _,v in pairs(methods) do
    if v == request_method then
      return true
    end
  end
  return false
end

```

Reply:
Hi Eloy thanks for the review comments, I'm working to resolve them.
One question: should't _if methods == nil **and** next(methods) == nil  then_
be: _if methods == nil **or** next(methods) == nil  then_
(So or instead of and) because in the case of and when methods == nil next(methods) will throw an error? Or am I missing something here?

Reply:
rest is changed